shell/app/modules/project/pages/settings/components/workflow-setting/branch-policy.tsx

```diff
@@ -0,0 +1,818 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// This program is free software: you can use, redistribute, and/or modify
+// it under the terms of the GNU Affero General Public License, version 3
+// or later ("AGPL"), as published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+// FITNESS FOR A PARTICULAR PURPOSE.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+import React from 'react';
+import { getBranchPolicy, updateBranchPolicy, BranchPolicy } from 'project/services/project-workflow';
+import { useMount } from 'react-use';
+import { EmptyHolder, ErdaIcon, Ellipsis } from 'common';
+import { Button, Switch, Input, Select, Popconfirm, message, Tooltip, Spin } from 'antd';
+import { uuid } from 'common/utils';
+import { throttle, compact } from 'lodash';
+import { FlowType } from 'project/common/config';
+import ResizeObserver from 'rc-resize-observer';
+import { branchNameValidator, branchNameWithoutWildcard } from 'project/common/config';
+import { WithAuth } from 'user/common';
+import themeColor from 'app/theme-color.mjs';
+import i18n from 'i18n';
+interface IProps {
+  projectId: string;
+  editAuth: boolean;
+}
+
+interface BranchPolicyData extends BranchPolicy {
+  id: string;
+  openTempMerge: boolean;
+}
+
+export const convertPolicyData = (d: BranchPolicy[]) =>
+  d.map((item) => {
+    return {
+      ...item,
+      id: uuid(),
+      openTempMerge: !!item?.policy?.tempBranch,
+      policy: item.policy && {
+        ...item.policy,
+        targetBranch: item.policy.targetBranch && {
+          ...item.policy.targetBranch,
+          cherryPick: item.policy.targetBranch.cherryPick || undefined,
```

Comment:
is `|| undefined` required

Reply:
yes, cherrypick maybe "",  initial value needs to split(','),  ''.split(',') = [''], the expected value is []。
after initial  cherrypick could be ''